README.md

```diff
@@ -1,7 +1,334 @@
+```csharp
+string documentId = "https://coveo.com";
+client.DocumentManager.DeleteDocument(sourceId, documentId, null);
+```
+**Good to know:**
+* Prefer deleting documents in batches to save API calls and to have a better application performance.
+
+### Delete a batch of documents
+```csharp
+IList<string> documentsIdstoDelete = new List<string> {
+    "https://coveo.com",
+    "https://coveo.com/a page"
+};
+client.DocumentManager.DeleteDocuments(sourceId, documentsIdstoDelete, null);
+```
+
+### Delete a specific document and its children
+You can delete a specific document and all its children easily. For more information about deleting a document and its children, visit [Deleting an Item and Optionally, its Children in a Push Source](https://docs.coveo.com/en/171/cloud-v2-developers/deleting-an-item-and-optionally-its-children-in-a-push-source)
+
+In this example, imagine you have added a document with an ID of `http://coveo.com/parent` and another document with an ID of `http://coveo.com/parent/child`. Clearly, it this example, the second document is the child of the former. To delete a document and all its children, it is pretty easy:
+```csharp
+string parentDocumentId = "http://coveo.com/parent";
+client.DocumentManager.DeleteDocument(sourceId, parentDocumentId, null, true);
+```
+
+### Delete documents older than a specific ordering ID
+Remember the ordering ID? You can send a request to your Push API source to delete all documents that were pushed with an ordering ID less than the one you provide.
+```csharp
+ulong orderingId = 12345; // Every document in the source that has an ordering lower than 12345 will be deleted.
+client.DocumentManager.DeleteDocumentsOlderThan(sourceId, orderingId, null);
+```
+**Good to know:**
+* The third argument in `DeleteDocumentsOlderThan` is the processing delay. When passing `null`, it uses the default value of 15 minutes. For more information about processing delay, visit [QueueDelay](https://docs.coveo.com/en/131/cloud-v2-developers/deleting-old-items-in-a-push-source). 
+
+### Combine delete and add
+You can combine a batch of documents to be deleted and a batch of documents to be added in the same call.
+```csharp
+PushDocument firstDocumentToAdd = new PushDocument("http://www.coveo.com/page/child") {
+    ClickableUri = "http://www.coveo.com/page/child",
+    ModifiedDate = DateTime.UtcNow
+};
+
+PushDocument secondDocumentToAdd = new PushDocument("http://www.coveo.com/page/child/child") {
+    ClickableUri = "http://www.coveo.com/page/child/child",
+    ModifiedDate = DateTime.UtcNow
+};
+
+IList<PushDocument> documentsToAdd = new List<PushDocument> {
+    firstDocumentToAdd,
+    secondDocumentToAdd
+};
+
+IList<string> documentsToDelete = new List<string> {
+    "http://coveo.com/parent/child",
+    "http://coveo.com/parent"
+};
+
+client.DocumentManager.AddOrUpdateDocuments(sourceId, documentsToAdd, documentsToDelete, null);
+```
+
+## Adding securities to your documents
```

Comment:
securities -> permissions
we should keep using the standard terminology, or it quickly becomes very confusing to new users.
(applies to other occurrences as well)

Reply:
```suggestion
## Adding permissions to your documents
```